File: app/src/main/java/com/lzm/lightLive/http/request/hy/HyRecommendConverter.java
package com.lzm.lightLive.http.request.hy;

import com.lzm.lightLive.http.CustomGsonConvertFactory;
import java.io.IOException;
import okhttp3.ResponseBody;

public class HyRecommendConverter extends CustomGsonConvertFactory.ResponseConverter {

    @Override
    public String convertResponse(ResponseBody value) throws IOException {
        String valueStr = value.string();
        return valueStr.substring("getLiveListJsonpCallback(".length(), valueStr.length() - 1);
    }

}

File: app/src/main/java/com/lzm/lightLive/http/request/hy/HyRecommendConverter.kt
package com.lzm.lightLive.http.request.hy

import com.lzm.lightLive.http.CustomGsonConvertFactory.ResponseConverter
import okhttp3.ResponseBody
import java.io.IOException

class HyRecommendConverter : ResponseConverter() {
    @Throws(IOException::class)
    override fun convertResponse(value: ResponseBody): String {
        val valueStr = value.string()
        return valueStr.substring("getLiveListJsonpCallback(".length, valueStr.length - 1)
    }
}